plugins/rebalance/rest.go

```diff
@@ -0,0 +1,242 @@
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+
+	rebalinfo, err := GetRebalanceInfo(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check rebalance process is started or not
+	if rebalinfo.Status != rebalanceapi.GfDefragStatusStarted {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Rebalance process is not started on this volume",
+			api.ErrCodeDefault)
+		return
+	}
+
+	// Check remove brick operation is running
+	//TODO
+
+	//A simple transaction to stop rebalance
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "rebal-stop.Commit",
+			Nodes:  txn.Nodes,
+		},
+		{
+			DoFunc: "rebal-stop.StoreVolume",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		unlock,
+	}
+
+	err = txn.Ctx.Set("volname", volname)
+	if err != nil {
+		logger.WithError(err).Error("failed to set request in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	rebalinfo.Volname = volname
+	rebalinfo.Status = rebalanceapi.GfDefragStatusStopped
+	rebalinfo.Cmd = rebalanceapi.GfDefragCmdStop
+
+	err = txn.Ctx.Set("rinfo", rebalinfo)
+	if err != nil {
+		logger.WithError(err).Error("failed to set rebalance info in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	err = txn.Do()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to stop rebalance on volume")
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Ctx.Logger().WithField("volname", rebalinfo.Volname).Info("rebalance stopped")
+	restutils.SendHTTPResponse(r.Context(), w, http.StatusOK, rebalinfo)
+}
+func rebalanceStatus(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	// collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	// Validate rebalance command
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.DistCount == 1 {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume is not distributed volume",
+			api.ErrCodeDefault)
+		return
+	}
+
+	rebal, err := GetRebalanceInfo(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
```

Comment:
Please add TODO here to get realtime status from all Volume nodes. Transaction and RPC communication is required for the same. Refer Geo-replication status to do this conditionally (if rebal.Status == rebalanceapi.Started then collect realtime status from all Volume nodes)